widget-showcase/tests/unit/all.ts

```diff
@@ -0,0 +1,639 @@
+	'can change the time picker'() {
+		const widget = harness(App);
+
+		widget.callListener('onChange', {
+			key: 'time-picker',
+			args: ['on change value']
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'time-picker')!, {
+			value: 'on change value'
+		});
+
+		widget.expectRender(expectedWidget, 'The value is updated');
+
+		widget.callListener('onRequestOptions', {
+			key: 'time-picker',
+			args: ['', [{hello: 'world'}]]
+		});
+
+		assignProperties(findKey(expectedWidget, 'time-picker')!, {
+			options: [{hello: 'world'}]
+		});
+
+		widget.expectRender(expectedWidget, 'The options are updated');
+
+		widget.destroy();
+	},
+
+	'can change the select box'() {
+		const widget = harness(App);
+
+		widget.callListener('onChange', {
+			key: 'select',
+			args: [{
+				value: 'Hello there'
+			}]
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'select')!, {
+			value: 'Hello there'
+		});
+
+		widget.expectRender(expectedWidget, 'The select box is changed');
+		widget.destroy();
+	},
+
+	'can change the multi select'() {
+		const widget = harness(App);
+
+		widget.callListener('onChange', {
+			key: 'multiselect',
+			args: [{
+				value: 'New option value'
+			}]
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'multiselect')!, {
+			value: 'New option value'
+		});
+
+		widget.expectRender(expectedWidget, 'The multi select box is changed');
+		widget.destroy();
+	},
+
+	'can change the slider value'() {
+		const widget = harness(App);
+
+		widget.callListener('onInput', {
+			key: 'slider',
+			args: [{
+				target: {
+					value: '33.101'
+				}
+			}]
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'slider')!, {
+			value: 1
+		});
+
+		widget.expectRender(expectedWidget, 'The slider value is set');
+		widget.destroy();
+	},
+
+
+	'can change the text area value'() {
+		const widget = harness(App);
+
+		widget.callListener('onChange', {
+			key: 'text-area',
+			args: [{
+				target: {
+					value: 'some textarea value'
+				}
+			}]
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'text-area')!, {
+			value: 'some textarea value'
+		});
+
+		widget.expectRender(expectedWidget, 'The textarea value is set');
+		widget.destroy();
+	},
+
+	'can open and close a dialog'() {
+		const widget = harness(App);
+
+		widget.callListener('onClick', {
+			key: 'dialog-button'
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'dialog')!, {
+			open: true
+		});
+
+		// widget.callListener('onRequestClose', {
+		// 	key: 'dialog'
+		// });
+
+		widget.expectRender(expectedWidget, 'the dialog opens');
+		widget.destroy();
+	}
+});
```

Comment:
nit: new line at end of file

Reply:
Done! We should use linting for this one!